tests/embind_with_asyncify.cpp

```diff
@@ -0,0 +1,18 @@
+#include <assert.h>
+
+#include <string>
+
+#include <emscripten.h>
+#include <emscripten/val.h>
+
+using namespace emscripten;
+
+int main() {
+  val fetch = val::global("fetch");
+  std::string url = "data:text/plain,foo";
+  val async_response = fetch(url);
+  val response = async_response.await();
```

Comment:
Interesting. We see later that this could be written as `fetch(url).await();` to better match the JS-style `await fetch(url);`

Was this intentional to show how it works more clearly? Because it had that effect for me at least. Nice.

Reply:
Oh yeah, sorry - in fact, this entire snippet could be written as a single chain of calls, but for a test I find it better to split steps into separate variables for easier debugging when something goes wrong.